core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2323,31 +2323,75 @@ private[spark] object Utils
    * configure a new log4j level
    */
   def setLogLevel(l: Level): Unit = {
-    val (ctx, loggerConfig) = getLogContext
+    val (ctx, loggerConfig) = getLogContext()
     loggerConfig.setLevel(l)
     ctx.updateLoggers()
 
     // Setting threshold to null as rootLevel will define log level for spark-shell
     Logging.sparkShellThresholdLevel = null
   }
 
+  /**
+   * configure a new log4j level for specific package or class
+   *
+   * @param name package or class name such as "org.apache.spark" or
+   *             "org.apache.spark.SparkContext"
+   * @param level new level of [[org.apache.log4j.Level]]
+   */
+  def setLogLevel(name: String, level: Level): Unit = {
```

Comment:
I don't think we need this override, do we? the original method didn't support this.
Id also imagine that the old method just just call this one with some value representing the root logger

Reply:
Do you want me to remove old method `def setLogLevel(l: Level): Unit`, and calling new method instead?

Reply:
Oh, nevermind, there were already two overloads? then two new overloads makes sense.
But yeah they should call each other in a logical way